src/main/java/com/hms/flexyosisoftconnector/configuration/OSIsoftTagNamingScheme.java

```diff
@@ -130,12 +130,17 @@ public static String extractTagNameFromComponents(String tagName, String tagType
     String parsedTagName = "";
     final int lengthOfSingleOption = 2;
 
-    if (shortHandOSIsoftNameOptionSelected.length() <= lengthOfSingleOption) {
+    if (shortHandOSIsoftNameOptionSelected.length() >= lengthOfSingleOption) {
       // loop through every option present
       for (int index = 0;
           index < shortHandOSIsoftNameOptionSelected.length();
           index += lengthOfSingleOption) {
-        String shortHandOption = shortHandOSIsoftNameOptionSelected.substring(index, index + 1);
+
+        // add a optionNameDeliminator between name components in the OSIsoft tag name
+        if (index > 0) {
+          parsedTagName += optionNameDeliminator;
+        }
+        String shortHandOption = shortHandOSIsoftNameOptionSelected.substring(index, index + 2);

```

Comment:
magic number

Reply:
Not using lengthOfSingleOption here is why there was an issue in the first place.. fixed